File: easy/tree/Univalued Binary Tree.cpp
class Solution {
public:
    bool check(TreeNode* node,int val){
        if(node == nullptr) return true;
        if(node->val != val) return false;
        return check(node->left, val) && check(node->right, val);
    }
    bool isUnivalTree(TreeNode* root) {
        return check(root,root->val);
    }
};

File: easy/tree/Univalued Binary Tree.py
class Solution(object):
    def isUnivalTree(self, root):
        def check(node,val):
            if not node:
                return True
            if node.val != val:
                return False
            return check(node.left,val) and check(node.right,val)
        
        return check(root,root.val)
        
